internal/feature.go

```diff
@@ -68,6 +89,33 @@ func NewFeatureTest(name, version string, fn FeatureTestFn) func() error {
 	return ft.execute
 }
 
+// NewPortableFeatureTest creates a cross-platform [FeatureTest].
+//
+// versions are in the format "GOOS:Major.Minor[.Patch]"
+func NewPortableFeatureTest(name string, fn FeatureTestFn, versions ...string) func() error {
```

Comment:
I'm not sure why this needs to have a separate constructor. Couldn't we take `Linux:6.11` and `6.11` to mean the same thing for backwards compat? (or just rewrite all the ones we have) Making `version` variadic in `NewFeatureTest` isn't a huge problem as it only has a few dozen invocations. Refactoring should be fairly straightforward.